samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/ChatHistoryController.cs

```diff
@@ -171,11 +176,29 @@ public async Task<IActionResult> EditChatSessionAsync([FromBody] ChatSession cha
         }
 
         chat.Title = chatParameters.Title;
-        await this._chatSessionRepository.UpdateAsync(chat);
+        await this._chatSessionRepository.UpsertAsync(chat);
 
         return this.Ok(chat);
     }
 
+    /// <summary>
+    /// Service API to get a list of imported documents.
+    /// </summary>
+    [Authorize]
+    [Route("chatSession/{chatId:guid}/documents")]
```

Comment:
`sources` will be a better route than `documents`, IMO. (Sorry, I know it is my code :) )